src/system/group.cr

```diff
@@ -0,0 +1,136 @@
+require "c/grp"
+require "c/sys/limits"
+
+struct System::Group
+  {% if flag?(:darwin) || flag?(:openbsd) || flag?(:freebsd) %}
+    GID_MAX = LibC::GID_MAX
+  {% elsif flag?(:linux) %}
+    GID_MAX = 0xffffffff_u32
+  {% else %}
+    GID_MAX = 0xffff_u32
+  {% end %}
+  NGROUPS_MAX = LibC::NGROUPS_MAX
+
+  # Returns the group name for the given group ID.
+  # Raises `System::Group::NotFoundError` if no user exists.
+  def self.name(gid : Int) : String
+    name?(gid) || raise NotFoundError.new("Group with gid #{gid.inspect} was not found")
+  end
+
+  # Returns the group name for the given group ID.
+  # Returns `nil` if no group exists.
+  def self.name?(gid : Int) : String?
+    check_gid_in_bounds(gid)
+    group_struct = LibC.getgrgid(gid)
+    return String.new(group_struct.value.gr_name) if group_struct
+    nil
```

Comment:
redundant `nil`

Reply:
This seems like a readability nightmare. Is this really the standard...

Reply:
I would have written it like:

```crystal
group_struct ? String.new(group_struct.value.gr_name) : nil
```

Reply:
Yeah it's a readability nightmare, which is why you reverse the condition.

Either use what @asterite said or:
```cr
return nil unless group_struct
String.new(group_struct.value.gr_name)
```